.github/workflows/tests.yml

```diff
@@ -64,71 +63,137 @@ jobs:
 
             if (label_names.includes('CI-test-bot-fail-fast')) {
               console.log('CI-test-bot-fail-fast label found. Passing --fail-fast to brew test-bot.')
-              console.log('::set-output name=test-bot-fail-fast::--fail-fast');
+              core.setOutput('test-bot-fail-fast', '--fail-fast')
             } else {
               console.log('No CI-test-bot-fail-fast label found. Not passing --fail-fast to brew test-bot.')
-              console.log('::set-output name=test-bot-fail-fast::');
+              core.setOutput('test-bot-fail-fast', '')
             }
 
             if (label_names.includes('CI-test-bot-skip-recursive-dependents')) {
               console.log('CI-test-bot-skip-recursive-dependents label found. Passing --skip-recursive-dependents to brew test-bot.')
-              console.log('::set-output name=test-bot-skip-recursive-dependents::--skip-recursive-dependents');
+              core.setOutput('test-bot-skip-recursive-dependents', '--skip-recursive-dependents')
             } else {
               console.log('No CI-test-bot-skip-recursive-dependents label found. Not passing --skip-recursive-dependents to brew test-bot.')
-              console.log('::set-output name=test-bot-skip-recursive-dependents::');
+              core.setOutput('test-bot-skip-recursive-dependents', '')
+            }
+
+            /* RUNNERS */
+
+            const runners = ["11-arm64", "11", "10.15", "10.14"]
+
+            if (label_names.includes('CI-linux-self-hosted')) {
+              runners.push('linux-self-hosted-1')
+            } else {
+              runners.push('ubuntu-latest')
             }
 
+            core.setOutput('runners', JSON.stringify(runners))
+
   tests:
-    needs: tap_syntax
-    if: github.event_name == 'pull_request' && needs.tap_syntax.outputs.syntax-only == 'false'
+    needs: pre_tests
+    if: github.event_name == 'pull_request' && needs.pre_tests.outputs.syntax-only == 'false'
     strategy:
       matrix:
-        version: ["11-arm64", "11", "10.15", "10.14"]
+        runner: ${{fromJson(needs.pre_tests.outputs.runners)}}
       fail-fast: false
-    runs-on: ${{ matrix.version }}
+    runs-on: ${{matrix.runner}}
     timeout-minutes: 4320
     env:
-      PATH: "/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
-      HOMEBREW_DEVELOPER: 1
-      GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED: 1
       GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
       HOMEBREW_GITHUB_API_TOKEN: ${{secrets.GITHUB_TOKEN}}
     steps:
+      - name: Set environment variables
+        run: |
+          if [ "$RUNNER_OS" = 'macOS' ]; then
+            echo 'PATH=/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin' >> $GITHUB_ENV
+            echo 'GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED=1' >> $GITHUB_ENV
+          else
+            echo 'HOMEBREW_CORE_GIT_REMOTE=${{github.event.repository.html_url}}' >> $GITHUB_ENV
+            echo 'HOMEBREW_FORCE_HOMEBREW_ON_LINUX=1' >> $GITHUB_ENV
+          fi
```

Comment:
- Why no path set on Linux? May also be worth having a different path for ARM/x86_64
- Maybe worth changing this `GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED` if we're not setting it on Linux when using a self-hosted worker?

Reply:
- we already set it in Dockerfile
- I'm open for discussion

Reply:
> * we already set it in Dockerfile

Gotcha 👍🏻. Worth a comment? I don't feel strongly either way.

> * I'm open for discussion

🆒. think it'd be good to see this! Have opened https://github.com/Homebrew/actions/pull/198 and https://github.com/Homebrew/homebrew-test-bot/pull/623 to allow this variable to be changed in a follow-up PR (once they are merged).

Reply:
I don't really think we need a comment on this PATH thing.